### PR TITLE
feat: jwt-plugin support custom parameters

### PR DESCRIPTION
**Pull request by liangliang4ward**
### What this PR does / why we need it:

jwt-plugin support custom parameters

Fix https://github.com/apache/apisix/issues/5710
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [x] Have you modified the corresponding document?
* [x] Is this PR backward compatible? **If it is not backward compatible, please discuss on the [mailing list](https://github.com/apache/apisix/tree/master#community) first**

### CODE REVIEW

File: docs/en/latest/plugins/jwt-auth.md [hunk 1]
```diff
@@ -46,6 +48,14 @@ For more information on JWT, refer to [JWT](https://jwt.io/) for more informatio
 
 **Note**: To enable vault integration, first visit the [config.yaml](https://github.com/apache/apisix/blob/master/conf/config.yaml) update it with your vault server configuration, host address and access token. You can take a look of what APISIX expects from the config.yaml at [config-default.yaml](https://github.com/apache/apisix/blob/master/conf/config-default.yaml) under the vault attributes.
 
+For route side:
+
+| Name | Type   | Requirement | Default | Valid | Description                                                                  |
+| ---- | ------ | ----------- | ------- | ----- | ---------------------------------------------------------------------------- |
+| header  | string | optional    | authorization        |       | the header we get the token from |
+| query   | string | optional    | jwt        |       | the querystring we get the token from, which priority is lower than header |
+| cookie | string | optional    | jwt |       | the cookie we get the token from, which priority is lower than querystring |
```
Comment (spacewander) [inline docs/en/latest/plugins/jwt-auth.md:57]: Would be better to use "query string"?

File: t/plugin/jwt-auth2.t [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+
+=== TEST 5: verify (in query)
+--- request
+GET /hello?jwt-query=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJrZXkiOiJ1c2VyLWtleSIsImV4cCI6MTg3OTMxODU0MX0.fNtFJnNmJgzbiYmGB0Yjvm-l6A6M4jRV1l4mnVFSYjs
+--- response_body
+hello world
+
+
+
+=== TEST 6: verify (in header with out Bearer)
```
Comment (spacewander) [inline t/plugin/jwt-auth2.t:145]:
```suggestion
=== TEST 6: verify (in header without Bearer)
```

File: t/plugin/jwt-auth2.t [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+                [[{
+                    "username": "jack",
+                    "plugins": {
+                        "jwt-auth": {
+                            "key": "user-key",
+                            "secret": "my-secret-key"
+                        }
+                    }
+                }]],
+                [[{
```
Comment (spacewander) [inline t/plugin/jwt-auth2.t:56]: Could you remove the response data check?
We are going to remove them in other tests in https://github.com/apache/apisix/pull/6545

File: docs/en/latest/plugins/jwt-auth.md (hunk 1 above)
Comment (liangliang4ward) [reply docs/en/latest/plugins/jwt-auth.md:57]: key-auth.md use querystring,  I inherited it.

"query string" is good

File: t/plugin/jwt-auth2.t (hunk 3 above)
Comment (liangliang4ward) [reply t/plugin/jwt-auth2.t]: done

File: t/plugin/jwt-auth2.t (hunk 2 above)
Comment (liangliang4ward) [reply t/plugin/jwt-auth2.t]: done

File: docs/en/latest/plugins/jwt-auth.md (hunk 1 above)
Comment (liangliang4ward) [reply docs/en/latest/plugins/jwt-auth.md:57]: I can open a new PR to modify other files to achieve unification

File: apisix/plugins/jwt-auth.lua [hunk 4]
```diff
@@ -339,7 +352,7 @@ end
 
 
 function _M.rewrite(conf, ctx)
-    local jwt_token, err = fetch_jwt_token(ctx)
+    local jwt_token, err = fetch_jwt_token(conf,ctx)
```
Comment (leslie-tsang) [inline apisix/plugins/jwt-auth.lua:355]:
```suggestion
    local jwt_token, err = fetch_jwt_token(conf, ctx)
```

Comment (liangliang4ward) [reply apisix/plugins/jwt-auth.lua:355]: done

**Review by spacewander — APPROVED** [review]

**Review by leslie-tsang — APPROVED** [review]

**Pull request merged**